### PR TITLE
Run Cypress End to End

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR adds an end to end integration test using Cypress and MailSlurp.

### No mocks
When Cypress is running in `ete` mode it does not mock idapi; all requests are sent to the `code` environment.

### MailSlurp
This is a third party testing service that we are trialling. It allows the following functions:

`emptyInbox` - we use this to clear the inbox we are using at the start of any test so that we don't read stale emails.
`createInbox` - useful for new testing registrations
`waitForLatestEmail` - this is the function we use to read the email from the inbox that we hope was sent

## Reset password
We now test this end to end flow using live endpoints. We still have the existing mocked tests under `cypress-mocked` but this new, live test gives us more confidence.

## Why are the tests failing?
These tests will fail until we have AWS keys set as Github Secrets

### CODE REVIEW

**Comment by oliverlloyd** [comment]
[End to end test released](https://trello.com/c/txLG5ccY/3252-end-to-end-test-released)

File: cypress/integration/ete/reset_password.spec.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    const existing = {
+      email: '0298a96c-1028-4e3d-b943-a2b478c84dbd@mailslurp.com',
+      inbox: '0298a96c-1028-4e3d-b943-a2b478c84dbd',
+    };
+
+    it("changes the reader's password", () => {
+      cy.emptyInbox(existing.inbox).then(() => {
+        cy.intercept({
+          method: 'GET',
+          url: 'https://api.pwnedpasswords.com/range/*',
```
Comment (ob6160) [inline cypress/integration/ete/reset_password.spec.js:15]: general q about behaviour of this api endpoint in the app for my benefit -- is it used to stop people from changing their password to a compromised one?

Comment (ob6160) [reply cypress/integration/ete/reset_password.spec.js:15]: and-if it is-do we want a test case for when this happens?

**Review by ob6160 — APPROVED** [review]

File: cypress/integration/ete/reset_password.spec.js (hunk 1 above)
Comment (oliverlloyd) [reply cypress/integration/ete/reset_password.spec.js:15]: > general q about behaviour of this api endpoint in the app for my benefit -- is it used to stop people from changing their password to a compromised one?

Yes, exactly that. You can play around with this feature (and all of our pages and components) here: https://www.chromatic.com/component?appId=60929d168594f80039336501&name=Pages%2FChangePassword&buildNumber=631

Comment (oliverlloyd) [reply cypress/integration/ete/reset_password.spec.js:15]: > and-if it is-do we want a test case for when this happens?

Excellent point. This is defo something we want to be testing for, but actually [we already do](https://github.com/guardian/gateway/blob/main/cypress/integration/mocked/change_password.spec.js#L103)

File: .github/workflows/cypress-ete.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Cypress run
+        uses: cypress-io/github-action@v2
+        env:
+          PORT: 8861
+          NODE_ENV: development
+          ENVIRONMENT: production
+          STAGE: production
```
Comment (coldlink) [inline .github/workflows/cypress-ete.yml:16]: stage in this case would be what the guardian use

so either `DEV`, `CODE`, or `PROD`

think in this case you want `CODE`

Comment (oliverlloyd) [reply .github/workflows/cypress-ete.yml]: You have hawk eyesight!

Comment (coldlink) [reply .github/workflows/cypress-ete.yml]: Yeah that's right, `CODE` is pretty much what we call our staging/manual testing environments. In the `yml` file you can see what CODE urls look like, generally in the form `*.code.dev-theguardian.com`

File: cypress/integration/ete/reset_password.spec.js (hunk 1 above)
Comment (oliverlloyd) [reply cypress/integration/ete/reset_password.spec.js:15]: Yeah, Chromatic is fantastic. It's caught a _lot_ of issues, especially over on DCR.

Whenever you add a story it is automatically captured by Chromatic and then visual regression gets run on each push

File: .github/workflows/cypress-ete.yml (hunk 2 above)
Comment (oliverlloyd) [reply .github/workflows/cypress-ete.yml]: `CODE` is the name for the test/stage/pre prod/qa environment. It is all of those rolled into one. Not sure why it's called 'CODE' - I think there's a story there

See: https://m.code.dev-theguardian.com/uk

File: cypress/integration/ete/reset_password.spec.js (hunk 1 above)
Comment (ob6160) [reply cypress/integration/ete/reset_password.spec.js:15]: visual regression checking is super neat, does a regression fail the build or is it mainly there for a quick check?

File: .github/workflows/cypress-ete.yml (hunk 2 above)
Comment (oliverlloyd) [reply .github/workflows/cypress-ete.yml]: > I'm going to be v annoying and ask loads of questions while I'm still new

ALWAYS ask questions - never annoying

File: cypress/integration/ete/reset_password.spec.js (hunk 1 above)
Comment (oliverlloyd) [reply cypress/integration/ete/reset_password.spec.js:15]: It fails the build if it finds diffs. Someone has to click through to Chromatic and manually approve them (or fix the bug which caused them)

Comment (ob6160) [reply cypress/integration/ete/reset_password.spec.js:15]: nice – sounds like a good setup :)

File: .github/workflows/cypress-ete.yml (hunk 2 above)
Comment (ob6160) [reply .github/workflows/cypress-ete.yml]: lovely, thanks guys

**Pull request merged**